src/python/WMCore/HTTPFrontEnd/RequestManager/ReqMgrRESTModel.py

```diff
@@ -272,7 +272,10 @@ def getRequest(self, requestName=None):
         if requestName == None:
             result = GetRequest.getRequests()
         else:
-            result = Utilities.requestDetails(requestName)
+            try:
+                result = Utilities.requestDetails(requestName)
+            except: # if we fail to get a request return nothing
```

Comment:
I see so this is not just for pycurl patch but current code as well. Yes I think it is better to catch 404 error other error might be server side error.